src/virtual-machine.js

```diff
@@ -559,10 +559,12 @@ class VirtualMachine extends EventEmitter {
             .then(validatedInput => {
                 const projectVersion = validatedInput[0].projectVersion;
                 if (projectVersion === 2) {
-                    return this._addSprite2(validatedInput[0], validatedInput[1]);
+                    return this._addSprite2(validatedInput[0], validatedInput[1])
+                        .then(() => this.runtime.emitProjectChanged());
                 }
                 if (projectVersion === 3) {
-                    return this._addSprite3(validatedInput[0], validatedInput[1]);
+                    return this._addSprite3(validatedInput[0], validatedInput[1])
+                        .then(() => this.runtime.emitProjectChanged());
```

Comment:
Since we reject in case of neither of these, can't we put this in the next `then()`? I.e. chained to `.then(validatedInput =>...).then(/* here? */)`?

Reply:
Yep, that makes sense! Good catch!

Reply:
Done!